specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2024-11-01/fluxconfiguration.json

```diff
@@ -0,0 +1,1860 @@
+  },
+  "definitions": {
+    "ScopeDefinition": {
+      "description": "Scope at which the configuration will be installed.",
+      "type": "string",
+      "enum": [
+        "cluster",
+        "namespace"
+      ],
+      "default": "cluster",
+      "x-ms-enum": {
+        "name": "ScopeType",
+        "modelAsString": true
+      }
+    },
+    "SourceKindDefinition": {
+      "description": "Source Kind to pull the configuration data from.",
+      "type": "string",
+      "enum": [
+        "GitRepository",
+        "Bucket",
+        "AzureBlob",
+        "OCIRepository"
+      ],
+      "x-ms-enum": {
+        "name": "SourceKindType",
+        "modelAsString": true
+      }
+    },
+    "GitRepositoryDefinition": {
+      "description": "Parameters to reconcile to the GitRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration git repository.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "repositoryRef": {
+          "description": "The source reference for the GitRepository object.",
+          "$ref": "#/definitions/RepositoryRefDefinition"
+        },
+        "sshKnownHosts": {
+          "description": "Base64-encoded known_hosts value containing public SSH keys required to access private git repositories over SSH",
+          "type": "string",
+          "x-nullable": true
+        },
+        "httpsUser": {
+          "description": "Plaintext HTTPS username used to access private git repositories over HTTPS",
+          "type": "string",
+          "x-nullable": true
+        },
+        "httpsCACert": {
+          "description": "Base64-encoded HTTPS certificate authority contents used to access git private git repositories over HTTPS",
+          "type": "string",
+          "x-nullable": true
+        },
+        "localAuthRef": {
+          "description": "Name of a local secret on the Kubernetes cluster to use as the authentication secret rather than the managed or user-provided configuration secrets.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "provider": {
+          "description": "Name of the provider used for authentication.",
+          "type": "string",
+          "x-nullable": true,
+          "enum": [
+            "azure",
+            "generic"
```

Comment:
Please make them 'Azure' and 'Generic' and add descriptions for each enum on what they mean. 

Descriptions are used to generate comments/docs and so end customer will have context on what this means.